vk-video/src/vulkan_decoder.rs

```diff
@@ -715,6 +715,11 @@ impl VulkanDecoder<'_> {
             }
         }
 
+        // this has to be done with Option and mut, because the closure we create has to be FnMut.
+        // this means we cannot consume its captures, so we have to take the option to be able to
+        // drop the resource.
```

Comment:
This is no longer true, as 24.0 includes my changes added after this comment. The closure can be `FnOnce` now, so there's no need to wrap it in `Some`. Same for all `DropCallbacks`.